### PR TITLE
[Refactor] Protobuf refactoring (DataModel feature branch)

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

Necessary changes related to DataModel refactoring: replacing `ZMGenericMessage` properties with `GenericMessage`.

## Dependencies
wireapp/wire-ios-data-model/pull/983

### CODE REVIEW

File: Sources/Request Strategies/Assets/AssetClientMessageRequestStrategyTests.swift [hunk 1]
```diff
@@ -258,7 +262,8 @@ class AssetClientMessageRequestStrategyTests: MessagingTestBase {
             XCTAssertNotNil(self.sut.nextRequest())
 
             // THEN
-            XCTAssertFalse(message.genericMessage!.content!.expectsReadConfirmation())
+            XCTAssertFalse( message.needsReadConfirmation)
+//            XCTAssertFalse(message.underlyingMessage!.content!.expectsReadConfirmation())
```
Comment (typfel) [inline Sources/Request Strategies/Assets/AssetClientMessageRequestStrategyTests.swift:266]: Commented out code

File: Sources/Request Strategies/Assets/AssetClientMessageRequestStrategyTests.swift [hunk 2]
```diff
@@ -244,7 +248,7 @@ class AssetClientMessageRequestStrategyTests: MessagingTestBase {
             XCTAssertNotNil(self.sut.nextRequest())
             
             // THEN
-            XCTAssertTrue(message.genericMessage!.content!.expectsReadConfirmation())
+           XCTAssertTrue( message.needsReadConfirmation)
```
Comment (typfel) [inline Sources/Request Strategies/Assets/AssetClientMessageRequestStrategyTests.swift:251]: I would keep checking the actual protobuf like before.

File: Sources/Request Strategies/Assets/AssetClientMessageRequestStrategyTests.swift [hunk 3]
```diff
@@ -258,7 +262,8 @@ class AssetClientMessageRequestStrategyTests: MessagingTestBase {
             XCTAssertNotNil(self.sut.nextRequest())
 
             // THEN
-            XCTAssertFalse(message.genericMessage!.content!.expectsReadConfirmation())
+            XCTAssertFalse( message.needsReadConfirmation)
```
Comment (typfel) [inline Sources/Request Strategies/Assets/AssetClientMessageRequestStrategyTests.swift:265]: I would keep checking the actual protobuf like before.

File: Sources/Request Strategies/Client Message/ClientMessageTranscoderTests.swift [hunk 4]
```diff
@@ -147,7 +147,7 @@ extension ClientMessageTranscoderTests {
             }
             
             // THEN
-            XCTAssertTrue(message.genericMessage!.content!.expectsReadConfirmation())
+            XCTAssertTrue( message.needsReadConfirmation)
```
Comment (typfel) [inline Sources/Request Strategies/Client Message/ClientMessageTranscoderTests.swift:150]: Same here, let's verify the actual protobuf.

File: Sources/Request Strategies/Client Message/ClientMessageTranscoderTests.swift [hunk 5]
```diff
@@ -168,7 +168,8 @@ extension ClientMessageTranscoderTests {
             }
             
             // THEN
-            XCTAssertFalse(message.genericMessage!.content!.expectsReadConfirmation())
+            XCTAssertFalse( message.needsReadConfirmation)
```
Comment (typfel) [inline Sources/Request Strategies/Client Message/ClientMessageTranscoderTests.swift:171]: Same here, let's verify the actual protobuf.

File: Sources/Request Strategies/Assets/AssetClientMessageRequestStrategyTests.swift (hunk 1 above)
Comment (KaterinaWire) [reply Sources/Request Strategies/Assets/AssetClientMessageRequestStrategyTests.swift]: Removed

File: Sources/Request Strategies/Assets/AssetClientMessageRequestStrategyTests.swift (hunk 2 above)
Comment (KaterinaWire) [reply Sources/Request Strategies/Assets/AssetClientMessageRequestStrategyTests.swift]: Done

File: Sources/Request Strategies/Assets/AssetClientMessageRequestStrategyTests.swift (hunk 3 above)
Comment (KaterinaWire) [reply Sources/Request Strategies/Assets/AssetClientMessageRequestStrategyTests.swift]: Done

File: Sources/Request Strategies/Client Message/ClientMessageTranscoderTests.swift (hunk 4 above)
Comment (KaterinaWire) [reply Sources/Request Strategies/Client Message/ClientMessageTranscoderTests.swift]: Done

File: Sources/Request Strategies/Client Message/ClientMessageTranscoderTests.swift (hunk 5 above)
Comment (KaterinaWire) [reply Sources/Request Strategies/Client Message/ClientMessageTranscoderTests.swift]: Done

**Pull request merged**